src/playlist.js

```diff
@@ -0,0 +1,424 @@
+
+    // Update currentIndex if inserting new elements earlier in the array than the current item
+    if (resolvedIndex <= this.currentIndex_) {
+      this.currentIndex_ += newItems.length;
+    }
+
+    this.trigger({
+      type: 'playlistadd',
+      count: newItems.length,
+      index: resolvedIndex
+    });
+
+    return [...newItems];
+  }
+
+  /**
+   * A custom DOM event that is fired when new item(s) are removed from the
+   * current playlist (rather than replacing the entire playlist).
+   *
+   * This is fired synchronously as it does not affect playback.
+   *
+   * @typedef  {Object} PlaylistRemoveEvent
+   * @see      [CustomEvent Properties]{@link https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent}
+   * @property {string} type - Always "playlistremove"
+   *
+   * @property {number} count - The number of items that were removed.
+   *
+   * @property {number} index - The starting index where item(s) were removed.
+   */
+
+  /**
+   * Removes a specified number of items from the playlist, starting at the given index.
+   * Adjusts the current index if it falls within the range of the removed items.
+   *
+   * @param {number} index - The starting index to remove items from. If out of bounds, no removal occurs.
+   * @param {number} [count=1] - The number of items to remove. Defaults to 1. Removal occurs only if count is a positive number.
+   * @return {Object[]} An array of the removed playlist items.
+   * @fires playlistremove - Triggered when items are successfully removed.
+   */
+  remove(index, count = 1) {
+    if (!isIndexInBounds(this.list_, index)) {
+      this.log_.error('Index is out of bounds.');
+      return [];
+    }
+
+    if (typeof count !== 'number' || count < 0) {
+      this.log_.error('Invalid count for removal.');
+      return [];
+    }
+
+    // Constrain the removal count to the number of items that are actually available to remove starting at that index
+    const actualCount = Math.min(count, this.list_.length - index);
+    const removedItems = this.list_.splice(index, actualCount);
+
+    this.adjustCurrentIndexAfterRemoval_(index, actualCount);
+
+    this.trigger({
+      type: 'playlistremove',
+      count: actualCount,
+      index
+    });
+
+    return [...removedItems];
+  }
+
+  /**
+   * Sorts the playlist array.
+   *
+   * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort}
+   * @param {Function} compare - A comparator function as per the native Array method.
+   * @fires playlistsorted - Triggered after the playlist is sorted internally.
+   */
+  sort(compare) {
+    if (!this.list_.length || typeof compare !== 'function') {
+      return;
+    }
+
+    const currentItem = this.getCurrentItem();
+
+    this.list_.sort(compare);
+
+    // Update the current index after sorting
+    this.currentIndex_ = this.list_.indexOf(currentItem);
+
+    this.trigger('playlistsorted');
```

Comment:
Could you please remind me why we want to trigger events in this class overall?

Reply:
Discussed this offline and we've decided to keep the events